### PR TITLE
add sample set md5 digest functionality for reproducibility

### PR DESCRIPTION
**Pull request by rdmorin**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [ ] I tested the new code for my use case (please provide a reproducible example of how you tested the new functionality)

- [ ] I ensured all dplyr functions that commonly conflict with other packages are fully qualified. 

This can be checked and addressed by running `check_functions.pl` and responding to the prompts. Test your code _after_ you do this.

- [ ] I generated the documentation and checked for errors relating to the new function (e.g. `devtools::document()`) and added `NAMESPACE` and all other modified files in the root directory and under `man`. 

### Optional but preferred with PRs

- [ ] I updated and/or successfully knitted a vignette that relies on the modified code (which ones?)

### Required

- [ ] I documented my function using [ROxygen style](https://jozef.io/r102-addin-roxytags/#:~:text=Inserting%20a%20skeleton%20%2D%20Do%20this,Shift%2BAlt%2BR%20).)

- [ ] All parameters for the function are described in the documentation and the function has a decriptive title. 

Example:
```
#' Use GISTIC2.0 scores output to reproduce maftools::chromoplot with more flexibility
#'
#' @param scores output file scores.gistic from the run of GISTIC2.0
#' @param genes_to_label optional. Provide a data frame of genes to label (if mutated). The first 3 columns must contain chromosome, start, and end coordinates. Another required column must contain gene names and be named `gene`. (truncated for example)
#' @param cutoff optional. Used to determine which regions to color as aberrant. Must be float in the range [0-1]. (truncated for example)
```

- [ ] My function uses a library that isn't already a dependency of GAMBLR and I made the package aware of this dependency using the function documentation `import` statment. 

Example:
```
#' @return nothing
#' @export
#' @import tidyverse ggrepel
```

## Checklist for changes to existing code

- [ ] I added/removed arguments to a function and updated documentation for all changed/new arguments

- [ ] I tested the new code for compatability with existing functionality in the Master branch (please provide a reprex of how you tested the original functionality)

### CODE REVIEW

**Comment by rdmorin** [comment]
This can't really be tested any more than it has been thus far until we start getting more updates to the sample sets. Can someone give it a review for obvious issues so we can get it into Master for more thorough testing later on?

File: R/utilities.R [hunk 1]
```diff
@@ -5,6 +5,100 @@ names(rainfall_conv) = c('A>G', 'T>C', 'C>T', 'G>A', 'A>T', 'T>A', 'A>C', 'T>G',
...
+
+}
+
+#' Generate an md5 hash for a set of samples to help ensure reproducibility
+#'
+#' @param these_samples_metadata Optionally provide a metadata table or any data frame with a column named sample_id that has been subset to the samples you're working with
+#' @param these_samples Optionally provide a vector of sample_id you are working with
+#' @param sample_set_name Optionall provide the name of a sample set in GAMBL and the function will load the samples from that set and provide the hash
+#' @return The md5 hash of the ordered set of sample_id
+#' @export
```
Comment (Kdreval) [inline R/utilities.R:71]: I think this needs the `#' @import` directive to add the digest and dplyr as dependencies?

File: R/utilities.R [hunk 2]
```diff
@@ -5,6 +5,100 @@ names(rainfall_conv) = c('A>G', 'T>C', 'C>T', 'G>A', 'A>T', 'T>A', 'A>C', 'T>G',
...
+      problems = dplyr::filter(to_check,md5_digest!=new_md5_digest)
+      print(problems)
+      stop("some md5 digests do not match. Have these sample sets changed???")
+      
+    }
+    
+  }else{
+    # just create a file that records the md5 digests for existing sample sets
+    sample_sets = suppressMessages(read_tsv(sample_sets_file))
+    set_names = select(sample_sets,-sample_id) %>% colnames()
```
Comment (mattssca) [inline R/utilities.R:50]: add `dplyr::` before `select`?

File: R/utilities.R [hunk 3]
```diff
@@ -5,6 +5,100 @@ names(rainfall_conv) = c('A>G', 'T>C', 'C>T', 'G>A', 'A>T', 'T>A', 'A>C', 'T>G',
...
+#' Generate an md5 hash for a set of samples to help ensure reproducibility
+#'
+#' @param these_samples_metadata Optionally provide a metadata table or any data frame with a column named sample_id that has been subset to the samples you're working with
+#' @param these_samples Optionally provide a vector of sample_id you are working with
+#' @param sample_set_name Optionall provide the name of a sample set in GAMBL and the function will load the samples from that set and provide the hash
+#' @return The md5 hash of the ordered set of sample_id
+#' @export
+#'
+#' @examples
+get_samples_md5_hash = function(these_samples_metadata,these_samples,sample_set_name,sample_sets_df){
```
Comment (mattssca) [inline R/utilities.R:74]: list `sample_sets_df` in the parameter descriptions?

**Review by mattssca — COMMENTED** [review]
Just a few small comments

File: R/utilities.R (hunk 3 above)
Comment (rdmorin) [reply R/utilities.R:74]: good call

File: R/utilities.R (hunk 2 above)
Comment (rdmorin) [reply R/utilities.R:50]: good catch!

**Review by Kdreval — APPROVED** [review]

**Pull request merged**